sympy/solvers/solveset.py

```diff
@@ -1754,6 +1757,50 @@ def _is_logarithmic(f, symbol):
     return rv
 
 
+def _is_lambert(f, symbol):
+    r"""
+    Return True if the equation is of Lambert type, else False.
+    Equations containing `Pow`, `log` or `exp` terms are currently
+    treated as Lambert types.
+    """
+    return any(isinstance(arg, (Pow, exp, log)) for arg in _term_factors(f))
+
+
+def _compute_lambert_solutions(lhs, rhs, symbol):
+    """
+    Computes the Lambert solutions. Returns `None` if it
+    fails doing so.
```

Comment:
Again can you be more precise? Is this solving equations that have LambertW? Or is solving equations whose solutions are given in terms of LambertW?

Is it possible to specify the general form of the kind of equation that this would solve.

Reply:
> Or is solving equations whose solutions are given in terms of LambertW?

Yes, this is the right one... I'll include that part also...

> Is it possible to specify the general form of the kind of equation that this would solve.

I'll think about it.